sdk/keyvault/azure-keyvault-certificates/azure/keyvault/certificates/_shared/async_client_base.py

```diff
@@ -107,3 +159,13 @@ def _build_pipeline(config: Configuration, transport: AsyncHttpTransport, **kwar
     @property
     def vault_url(self) -> str:
         return self._vault_url
+
+    async def __aenter__(self) -> "AsyncKeyVaultClientBase":
+        await self._client.__aenter__()
+        return self
+
+    async def __aexit__(self, *args) -> None:
+        await self.close(*args)
+
+    async def close(self, *args):
```

Comment:
`close` needs no parameters, and should have a simple docstring like "Closes sockets opened by the client".